mayastor/src/pool.rs

```diff
@@ -419,42 +420,60 @@ impl Pool {
                 return Ok(());
             }
         };
-        let base_bdev_type = base_bdev.driver();
-        debug!("Destroying bdev type {}", base_bdev_type);
+        if let Some(uri) = base_bdev.bdev_uri() {
+            debug!("destroying bdev {}", uri);
+            bdev_destroy(&uri)
+                .await
+                .map_err(|_e| Error::FailedDestroyBdev {
+                    bdev: base_bdev.name(),
+                    bdev_type: base_bdev.driver(),
+                    name,
+                    errno: -1,
+                })
+                .map(|_| Ok(()))?
+        } else {
+            let base_bdev_type = base_bdev.driver();
+            debug!(
+                "Destroying bdev {} type {}",
+                base_bdev.name(),
+                base_bdev_type
+            );
 
-        let (sender, receiver) = oneshot::channel::<i32>();
-        if base_bdev_type == "aio" {
-            unsafe {
-                bdev_aio_delete(
-                    base_bdev.as_ptr(),
-                    Some(done_cb),
-                    cb_arg(sender),
-                );
+            let (sender, receiver) = oneshot::channel::<i32>();
```

Comment:
Do we need this "legacy" stuff or is this just a stop gap until moac creates pools with URIs?

Reply:
Right. We will remove the "legacy" stuff altogether but later one, but for that the whole pool creation would need to change somewhat, beginning with examining the bdevs. I want to postpone that until later.